commit 72431acf31ae01369b0a7f1da503269fc0e81a46
Author: Sakura286 <chenxuan@iscas.ac.cn>
Date:   Mon Jan 19 09:30:55 2026 +0800

    riscv64 files

diff --git a/ml/backend/ggml/ggml/src/ggml-cpu/arch/riscv/riscv.go b/ml/backend/ggml/ggml/src/ggml-cpu/arch/riscv/riscv.go
new file mode 100644
index 0000000..00c5e6d
--- /dev/null
+++ b/ml/backend/ggml/ggml/src/ggml-cpu/arch/riscv/riscv.go
@@ -0,0 +1,5 @@
+package riscv
+	
+	// #cgo CXXFLAGS: -std=c++17
+	// #cgo CPPFLAGS: -I${SRCDIR}/../.. -I${SRCDIR}/../../.. -I${SRCDIR}/../../../../include
+	import "C"
\ No newline at end of file
diff --git a/ml/backend/ggml/ggml/src/ggml-cpu/cpu_riscv64.go b/ml/backend/ggml/ggml/src/ggml-cpu/cpu_riscv64.go
new file mode 100644
index 0000000..539211f
--- /dev/null
+++ b/ml/backend/ggml/ggml/src/ggml-cpu/cpu_riscv64.go
@@ -0,0 +1,3 @@
+package cpu
+	
+	import _ "github.com/ollama/ollama/ml/backend/ggml/ggml/src/ggml-cpu/arch/riscv"
\ No newline at end of file