From b976f614ed2ce3bf98f15e9a93761aafe15ba5a9 Mon Sep 17 00:00:00 2001
From: Sakura286 <chenxuan@iscas.ac.cn>
Date: Mon, 16 Mar 2026 15:26:02 +0800
Subject: [PATCH] disable httpmuxgo121 on newer version of go

---
 main.go | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/main.go b/main.go
index 650e03a..9a343f3 100644
--- a/main.go
+++ b/main.go
@@ -1,3 +1,5 @@
+//go:debug httpmuxgo121=0
+
 package main
 
 import (
-- 
2.53.0

