From 85aee3e710288343eca393272a418483ac547b83 Mon Sep 17 00:00:00 2001
From: Sakura286 <chenxuan@iscas.ac.cn>
Date: Tue, 24 Mar 2026 10:11:07 +0800
Subject: [PATCH] limit batch size to stablize

---
 api/types.go | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/api/types.go b/api/types.go
index 3ccf3ce..cba3ffd 100644
--- a/api/types.go
+++ b/api/types.go
@@ -896,7 +896,7 @@ func DefaultOptions() Options {
 		Runner: Runner{
 			// options set when the model is loaded
 			NumCtx:    int(envconfig.ContextLength()),
-			NumBatch:  512,
+			NumBatch:  8,
 			NumGPU:    -1, // -1 here indicates that NumGPU should be set dynamically
 			NumThread: 0,  // let the runtime decide
 			UseMMap:   nil,
-- 
2.53.0

