diff -ruN clr.orig/rocclr/device/rocm/rocvirtual.cpp clr/rocclr/device/rocm/rocvirtual.cpp
--- clr.orig/rocclr/device/rocm/rocvirtual.cpp	2026-06-29 16:22:57.199010359 +0800
+++ clr/rocclr/device/rocm/rocvirtual.cpp	2026-06-29 16:22:57.212150803 +0800
@@ -3582,7 +3582,11 @@
   std::memcpy(dst, src, size);
 
   // Add memory fence
-  _mm_sfence();
+  #if !defined(__riscv)
+          _mm_sfence();
+#else
+          asm volatile("fence w,w" ::: "memory");
+#endif
 #else
   std::memcpy(dst, src, size);
 #endif
@@ -3842,7 +3846,11 @@
       } else if (kernArgImpl == KernelArgImpl::DeviceKernelArgsReadback && argSize != 0) {
         _mm_sfence();
         *(argBuffer + argSize - 1) = *(parameters + argSize - 1);
-        _mm_mfence();
+        #if !defined(__riscv)
+          _mm_mfence();
+#else
+          asm volatile("fence rw,rw" ::: "memory");
+#endif
         auto kSentinel = *reinterpret_cast<volatile unsigned char*>(argBuffer + argSize - 1);
       }
     }
diff -ruN clr.orig/rocclr/os/os_posix.cpp clr/rocclr/os/os_posix.cpp
--- clr.orig/rocclr/os/os_posix.cpp	2026-06-29 16:22:57.199328299 +0800
+++ clr/rocclr/os/os_posix.cpp	2026-06-29 16:22:57.212276812 +0800
@@ -677,6 +677,7 @@
       LP64_SWITCH("movl %%esp", "movq %%rsp") ",%0"
       : "=r"(value)
 #elif defined(ATI_ARCH_ARM)
+#elif defined(ATI_ARCH_RISCV)
       "mov %0,sp"
       : "=r"(value)
 #else
