From 46c3e4bc94a25d5f3a19a884eaa8ab89156ceb52 Mon Sep 17 00:00:00 2001
From: johnny <johnnyychiu@gmail.com>
Date: Sat, 13 Jun 2026 15:51:19 +0100
Subject: [PATCH] Fix clang spinloop mwaitx include

Signed-off-by: johnny <johnnyychiu@gmail.com>
---
 csrc/spinloop.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/csrc/spinloop.cpp b/csrc/spinloop.cpp
index c29e48a5f0ec..3285b9a3ded5 100644
--- a/csrc/spinloop.cpp
+++ b/csrc/spinloop.cpp
@@ -7,7 +7,7 @@ extern "C" {
 
 #if defined(__i386__) || defined(__x86_64__)
   #include <cpuid.h>
-  #include <mwaitxintrin.h>
+  #include <x86intrin.h>
 #endif
 
 #if defined(CLOCK_MONOTONIC_RAW)
